lib/action.ml

```diff
@@ -357,28 +354,42 @@ module Action (Github_api : Api.Github) (Slack_api : Api.Slack) (Buildkite_api :
       let notifs = List.map (generate_commit_comment_notification ~slack_match_func api_commit n) channels in
       Lwt.return notifs
     | Status n ->
-      let%lwt channels = partition_status ctx n in
-      let%lwt slack_user_id =
-        match Util.Build.is_failed_build n with
-        | false -> Lwt.return_none
-        | true ->
-          let email = n.commit.commit.author.email in
-          (match%lwt Slack_api.lookup_user ~ctx ~cfg ~email () with
-          | Ok (res : Slack_t.lookup_user_res) -> Lwt.return_some res.user.id
-          | Error e ->
-            log#warn "couldn't match commit email %s to slack profile: %s" email e;
-            Lwt.return_none)
-      in
-      let%lwt failed_steps =
-        let repo_state = State.find_or_add_repo ctx.state repo.url in
-        match Util.Build.notify_fail n cfg with
-        | false -> Lwt.return_none
-        | true ->
-          let%lwt failed_steps = Util.Build.new_failed_steps ~get_build:(Buildkite_api.get_build ~ctx) n repo_state in
-          Lwt.return_some failed_steps
-      in
-      let notifs = List.map (generate_status_notification ?slack_user_id ?failed_steps cfg n) channels in
-      Lwt.return notifs
+      (try%lwt
+         let%lwt channels = partition_status ctx n in
+         let%lwt slack_user_id =
+           match Util.Build.is_failed_build n with
+           | false -> Lwt.return_none
+           | true ->
+             let email = n.commit.commit.author.email in
+             (match%lwt Slack_api.lookup_user ~ctx ~cfg ~email () with
+             | Ok (res : Slack_t.lookup_user_res) -> Lwt.return_some res.user.id
+             | Error e ->
+               log#warn "couldn't match commit email %s to slack profile: %s" email e;
+               Lwt.return_none)
+         in
+         let%lwt failed_steps =
+           let repo_state = State.find_or_add_repo ctx.state repo.url in
+           match Util.Build.notify_fail n cfg with
+           | false -> Lwt.return_none
+           | true ->
+             let%lwt failed_steps =
+               Util.Build.new_failed_steps ~get_build:(Buildkite_api.get_build ~ctx) n repo_state
+             in
+             Lwt.return_some failed_steps
+         in
+         let notifs = List.map (generate_status_notification ?slack_user_id ?failed_steps cfg n) channels in
```

Comment:
no changes on the code, just moved inside the `try%lwt` block. 

Below this part, it's new code.